server/src/utils.ts

```diff
@@ -39,19 +39,25 @@ export let findProjectRootOfFile = (
 };
 
 export let findBinaryFromProjectRoot = (
-  source: p.DocumentUri
+  projectRootPath: p.DocumentUri // This must be a directory and not a file!
 ): null | p.DocumentUri => {
-  let dir = path.dirname(source);
-  let bscNativeReScriptPath = path.join(dir, c.nodeModulesBinDir);
+  let bscNativeReScriptPath = path.join(
+    projectRootPath,
+    c.nodeModulesBinDir,
+    c.bscBinName
+  );
 
   if (fs.existsSync(bscNativeReScriptPath)) {
-    return bscNativeReScriptPath;
-  } else if (dir === source) {
+    return path.dirname(bscNativeReScriptPath);
```

Comment:
This does not look nice.
First create a path, then strip the last element.
Another recipe for future bugs.